sw/device/tests/crypto/cryptotest/firmware/ecdsa.c

```diff
@@ -0,0 +1,187 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/status.h"
+#include "sw/device/lib/crypto/impl/ecc/ecdsa_p256.h"
+#include "sw/device/lib/crypto/impl/ecc/p256_common.h"
+#include "sw/device/lib/crypto/impl/integrity.h"
+#include "sw/device/lib/crypto/impl/keyblob.h"
+#include "sw/device/lib/crypto/include/datatypes.h"
+#include "sw/device/lib/crypto/include/ecc.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/test_framework/ujson_ottf.h"
+#include "sw/device/lib/ujson/ujson.h"
+#include "sw/device/tests/crypto/cryptotest/json/ecdsa_commands.h"
+
+status_t handle_ecdsa(ujson_t *uj) {
+  // Declare ECDSA parameter ujson deserializer types
+  cryptotest_ecdsa_operation_t uj_op;
+  cryptotest_ecdsa_hash_alg_t uj_hash_alg;
+  cryptotest_ecdsa_curve_t uj_curve;
+  cryptotest_ecdsa_message_t uj_message;
+  cryptotest_ecdsa_signature_t uj_signature;
+  cryptotest_ecdsa_coordinate_t uj_qx;
+  cryptotest_ecdsa_coordinate_t uj_qy;
+
+  // Deserialize ujson byte stream into ECDSA parameters
+  TRY(ujson_deserialize_cryptotest_ecdsa_operation_t(uj, &uj_op));
+  TRY(ujson_deserialize_cryptotest_ecdsa_hash_alg_t(uj, &uj_hash_alg));
+  TRY(ujson_deserialize_cryptotest_ecdsa_curve_t(uj, &uj_curve));
+  TRY(ujson_deserialize_cryptotest_ecdsa_message_t(uj, &uj_message));
+  TRY(ujson_deserialize_cryptotest_ecdsa_signature_t(uj, &uj_signature));
+  TRY(ujson_deserialize_cryptotest_ecdsa_coordinate_t(uj, &uj_qx));
+  TRY(ujson_deserialize_cryptotest_ecdsa_coordinate_t(uj, &uj_qy));
+
+  otcrypto_ecc_curve_type_t curve_type;
+  otcrypto_unblinded_key_t public_key;
+  p256_point_t pub_p256;
+  size_t digest_len;
+
+  otcrypto_word32_buf_t signature_mut;
+  ecdsa_p256_signature_t signature_p256;
+  switch (uj_curve) {
+    case kCryptotestEcdsaCurveP256:
+      curve_type = kOtcryptoEccCurveTypeNistP256;
```

Comment:
This case block is getting quite long. Can you extract this into a separate function, maybe like `setNistP256Params`?

Reply:
It's long because it uses/sets a lot of curve-specific variables. If we do this, we'd have to return a struct of all the curve-specific variables we set.